src/coreclr/src/jit/hwintrinsic.h

```diff
@@ -315,6 +316,41 @@ struct HWIntrinsicInfo
     }
 };
 
+#if defined(TARGET_XARCH)
+#define _CMP_EQ_OQ 0x00    /* Equal (ordered, nonsignaling) */
```

Comment:
This seems like strange naming. Did you copy/paste from some other location? Should there be a comment about which location?

Would it be better to give them a better name prefix than `_CMP`? Or put them all in an `enum` instead of using `#define`?

Reply:
This is the naming of the comparison values used by the C++ Hardware Intrinsics where they are already defines.
For Windows, they are already included/available, but on Unix they aren't available without including `immintrin.h` and enable AVX support in the compilation (which we can't do).

We use different names, with a proper enum, on the C# side, but document them mapping to the C++ names: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/X86/Enums.cs

Reply:
I think using the existing C++ names makes the most sense, but we can't easily use those names in anything other than a `#define` due to other code potentially already defining them with the same values.

Reply:
I added the following comment for now:
```
// These use the same names as defined by the C++ HWIntrinsics. They are
// already defined/available on Windows but aren't available on Unix
// without including immintrin.h and enabling AVX in the compilation, so
// we will just redefine them here with the same values.
```

Reply:
> I think using the existing C++ names makes the most sense, but we can't easily use those names in anything other than a #define due to other code potentially already defining them with the same values.

Can we "port" the `enum` from the C# file you pointed out and document the values in the way?
```suggestion
enum class FloatComparisonMode : int
{
  OrderedEqualNonSignaling = 0, // _CMP_EQ_OQ 
  OrderedLessThanSignaling = 1, // _CMP_LT_OS
...
}
#define _CMP_EQ_OQ 0x00    /* Equal (ordered, nonsignaling) */
```

Reply:
Changed to mirroring the FloatComparisonMode enum.

Reply:
Thanks - looks much more readable now!

Nit: we don't use xml comments in c++
```c++
    /// <summary> 
    /// _CMP_EQ_OQ 
    /// </summary>
```